client/layout/index.jsx

```diff
@@ -350,6 +359,13 @@ class Layout extends Component {
 	}
 }
 
+// This is a temporal solution that should be removed once the experiment is over.
+const LayoutWithExperimentAssignment = ( props ) => {
+	const [ , experimentAssignment ] = useExperiment( 'calypso_plans_wapuu_sales_agent_v0' );
+
+	return <Layout { ...props } wapuuSalesAgentExperimentAssignment={ experimentAssignment } />;
```

Comment:
What necessitates adding this right at the root of the `Layout` component? Can't we add it as a child component inside?

Reply:
I described this situation in the PR description:

> I had to add a HoC but the other option is to add the experiment INSIDE the provider, which seems not optimal in my opinion, as the component will be loaded and not displayed, vs not loaded and displayed at all.

If in your is better to delegate it into the children, I'll do so. But in such case, OdieContext will be always available, regarding the experiment being treatment or control

Reply:
This seems like a reasonable trade-off to me right now, but happy to revisit it.